conda/common/io.py

```diff
@@ -395,16 +395,43 @@ def backdown_thread_pool(max_workers=10):
     """Tries to create an executor with max_workers, but will back down ultimately to a single
     thread of the OS decides you can't have more than one.
     """
+    from concurrent.futures import _base  # These "_" imports are gross, but I don't think there's an alternative  # NOQA
+    from concurrent.futures.thread import _WorkItem
+
+    class CondaThreadPoolExecutor(ThreadPoolExecutor):
+
+        def submit(self, fn, *args, **kwargs):
```

Comment:
This is an exact reimplementation of the `submit()` method on the parent class, except I've added `try/except` around `self._adjust_thread_count()`.  Seems to me like this is the technically sound and correct solution.  I don't have any independent references though, and I'm frustrated it was so dirty to implement.

Reply:
Maybe if you included a link to the exact lines you copied on github it'd make it easy to see if it's changed indicating you need to update your modified version?